src/server/engine_shard_set.cc

```diff
@@ -390,7 +390,7 @@ void EngineShard::PollExecution(const char* context, Transaction* trans) {
     }
     ++stats_.ooo_runs;
 
-    bool keep = trans->RunInShard(this);
+    bool keep = trans->RunInShard(this, true);
```

Comment:
should_run is also set for `SUSPENDED_Q`. 
Why not be more precise and pass `trans_mask & Transaction::OUT_OF_ORDER` ?

Reply:
Moreover, to avoid the situation where both bits are set, I suggest to add 
```
 sd.local_mask |= SUSPENDED_Q;
sd.local_mask &= ~OUT_OF_ORDER;  /
```
in WatchInShard

Reply:
True... I have actually a parallel PR where I remove the unwatch hop and make all wakeups single hop 🙄